src/Common/FileCacheFactory.cpp

```diff
@@ -51,18 +52,32 @@ FileCachePtr FileCacheFactory::get(const std::string & cache_base_path)
     throw Exception(ErrorCodes::BAD_ARGUMENTS, "No cache found by path: {}", cache_base_path);
 }
 
-FileCachePtr FileCacheFactory::getOrCreate(
-    const std::string & cache_base_path, const FileCacheSettings & file_cache_settings)
+FileCachePtr FileCacheFactory::getOrCreate(const std::string & cache_base_path, const FileCacheSettings & file_cache_settings)
 {
     std::lock_guard lock(mutex);
 
     auto * cache_data = getImpl(cache_base_path, lock);
     if (cache_data)
         return cache_data->cache;
 
-    auto cache = std::make_shared<LRUFileCache>(cache_base_path, file_cache_settings);
-    caches.emplace(cache_base_path, CacheData(cache, file_cache_settings));
-    return cache;
+    if (file_cache_settings.cache_method == "ARC")
+    {
+        auto cache = std::make_shared<ARCFileCache>(cache_base_path, file_cache_settings);
+        caches.emplace(cache_base_path, CacheData(cache, file_cache_settings));
+        return cache;
+    }
+    else if (file_cache_settings.cache_method == "LRU")
+    {
+        auto cache = std::make_shared<LRUFileCache>(cache_base_path, file_cache_settings);
+        caches.emplace(cache_base_path, CacheData(cache, file_cache_settings));
+        return cache;
+    }
+    else /// set default with ARC
+    {
+        auto cache = std::make_shared<LRUFileCache>(cache_base_path, file_cache_settings);
```

Comment:
> The comment and the code diverge. Also I would rather throw exception in this `else` (like 'unknown cache method') because If the user did not specify any cache method, then it anyway will be ARC: `cache_method = config.getString(config_prefix + ".data_cache_method", "ARC");` so it is correct, otherwise we must throw exception.

okay